### PR TITLE
Feature/8025 dcc validation ux/8245 validation screen

### PR DESCRIPTION
**Pull request by ArturFriesen**
## Description
Implementation of the certificate validation input screen.

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-8245

### CODE REVIEW

File: src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings [hunk 1]
```diff
@@ -2843,3 +2843,37 @@ Die Angaben werden statistisch ausgewertet, sie werden nicht zu einem Profil ges
 "HealthCertificate_Validation_Error_TRY_AGAIN" = "Ein Fehler ist aufgetreten. Bitte versuchen Sie es später noch einmal oder kontaktieren Sie die technische Hotline über App-Informationen -> Technische Hotline.";
 
 "HealthCertificate_Validation_Error_NO_NETWORK" = "Ihre Internetverbindung wurde unterbrochen. Um die Gültigkeitsprüfung des Zertifikates abzuschließen, prüfen Sie die Verbindung und versuchen Sie es erneut.";
+
+/* Health Certificate Validation */
+
+"HealthCertificate_Validation_Title" = "Gültigkeit des Zertifikats";
```
Comment (NataliaLemmerth) [inline src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings:2849]:
```suggestion
"HealthCertificate_Validation_Title" = "Gültigkeit des Zertifikates";
```

**Review by NataliaLemmerth — CHANGES_REQUESTED** [review]
Can you please change "Zertifikats" to "Zertifikates"

**Review by pascal-brause — APPROVED** [review]

**Comment by ArturFriesen** [comment]
> Can you please change "Zertifikats" to "Zertifikates"

done

**Review by NataliaLemmerth — APPROVED** [review]
UA approved

File: src/xcode/ENA/ENAUITests/HealthCertificate/Validation/ENAUITests_42_ValidateHealthCertificate.swift [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+		// Tap on Date Time Selection
+		app.cells[AccessibilityIdentifiers.HealthCertificate.Validation.dateTimeSelection].waitAndTap()
+
+		snapshot("screenshot_certificate_validation_date_selection")
+	}
+}
+
+extension XCUIGestureVelocity {
+	public static let justATinyBit: XCUIGestureVelocity = 0.3
+}
```
Comment (nickguendling) [inline src/xcode/ENA/ENAUITests/HealthCertificate/Validation/ENAUITests_42_ValidateHealthCertificate.swift:78]: Is this used anywhere?

File: src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Validation/HealthCertificateValidationViewModel.swift [hunk 3]
```diff
@@ -14,28 +14,176 @@ final class HealthCertificateValidationViewModel {
...
+					.body(
+						text: AppStrings.HealthCertificate.Validation.body2,
+						accessibilityIdentifier: ""
+					),
+					.headline(
+						text: AppStrings.HealthCertificate.Validation.headline2,
+						accessibilityIdentifier: ""
+					),
+					.bulletPoint(text: AppStrings.HealthCertificate.Validation.bullet1),
+					.space(height: 8),
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Validation/HealthCertificateValidationViewModel.swift:71]: Just mentioning: there is a `spacing: .large` option on the `.bulletPoint` cell, the you could probably remove the space after them

File: src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Validation/Cells/CountrySelectionCell.swift [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+	private lazy var picker: UIPickerView = {
+		let picker = UIPickerView()
+		picker.translatesAutoresizingMaskIntoConstraints = false
+		picker.delegate = self
+		picker.tintColor = .enaColor(for: .tint)
+		return picker
+	}()
+
+	private lazy var seperator: UIView = {
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Validation/Cells/CountrySelectionCell.swift:98]:
```suggestion
	private lazy var separator: UIView = {
```

File: src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Validation/Cells/ValidationDateSelectionCell.swift [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+		} else if #available(iOS 13.4, *) {
+			datePicker.preferredDatePickerStyle = .wheels
+		}
+
+		datePicker.addTarget(self, action: #selector(didSelectDate(datePicker:)), for: .valueChanged)
+
+		return datePicker
+	}()
+
+	private lazy var seperator: UIView = {
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Validation/Cells/ValidationDateSelectionCell.swift:138]:
```suggestion
	private lazy var separator: UIView = {
```

**Review by nickguendling — CHANGES_REQUESTED** [review]
I checked it out and found small issues:
- The acceptance criteria state that:
  - The default for the first access is Germany
  - The selected country will be the default for subsequent calls of the screen
  - The default for the first access is the current date time
  - The selected date/time will be the default for subsequent calls of the screen
- I think it would make sense to sort the countries alphabetically, you could just use `.sortedByLocalizedName` on the `Country` array. Maybe we should check with Max or Martin.
- Germany is preselected but not on the Picker, when it is shown the first country is shown as selected.

File: src/xcode/ENA/ENAUITests/HealthCertificate/Validation/ENAUITests_42_ValidateHealthCertificate.swift (hunk 2 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENAUITests/HealthCertificate/Validation/ENAUITests_42_ValidateHealthCertificate.swift:78]: no, removed.

File: src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Validation/HealthCertificateValidationViewModel.swift (hunk 3 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Validation/HealthCertificateValidationViewModel.swift:71]: Good hint. Changed

File: src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Validation/Cells/CountrySelectionCell.swift (hunk 4 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Validation/Cells/CountrySelectionCell.swift:98]: changed

File: src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Validation/Cells/ValidationDateSelectionCell.swift (hunk 5 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Validation/Cells/ValidationDateSelectionCell.swift:138]: changed

**Comment by ArturFriesen** [comment]
> I checked it out and found small issues:
> 
> * The acceptance criteria state that:
>   
>   * The default for the first access is Germany
>   * The selected country will be the default for subsequent calls of the screen
>   * The default for the first access is the current date time
>   * The selected date/time will be the default for subsequent calls of the screen
> * I think it would make sense to sort the countries alphabetically, you could just use `.sortedByLocalizedName` on the `Country` array. Maybe we should check with Max or Martin.
> * Germany is preselected but not on the Picker, when it is shown the first country is shown as selected.

Good points. Fixed the preselection on the Picker. And sorted the countries.

**Review by nickguendling — APPROVED** [review]

**Pull request merged**